### ISSUE TITLE
NEARBY filter inaccurate when used with a geofence

### ISSUE THREAD

**Issue by stcktrce** [tidwall/tile38#431]
The results of `NEARBY` can differ when they are called from a geofence or directly.

# steps to reproduce

1. Open two `tile38-cli` instances
    - On one of the instances run `NEARBY fleet FENCE DETECT enter,inside,exit POINT 52.7563395 -0.8838196 200`
2. Run `SET fleet alpha POINT 52.7563759 -0.8856761`
3. Run `NEARBY fleet LIMIT 1 POINT 52.7563395 -0.8838196 200`
4. Observe as the last `NEARBY` call returns data, but the previously set geofence does not trigger an `inside` call.

We are seeing these issues on `1.15.0`

```
{"ok":true,"info":{"aof_current_rewrite_time_sec":0,"aof_enabled":1,"aof_last_rewrite_time_sec":0,"aof_rewrite_in_progress":0,"cluster_enabled":0,"connected_clients":139,"connected_slaves":0,"expired_keys":0,"redis_version":"1.15.0","tile38_version":"1.15.0","total_commands_processed":69241,"total_connections_received":17570,"uptime_in_seconds":1294284,"used_cpu_sys":851,"used_cpu_sys_children":0,"used_cpu_user":765,"used_cpu_user_children":0,"used_memory":22651840},"elapsed":"533.086µs"}
```

**Comment by stcktrce** [comment]
👍 Just tested it, looks resolved. Thanks a ton!

**Comment by tidwall** [comment]
You're welcome. I just published the v1.16.2 release to include this fix.
